test/cmd/apply.sh

```diff
@@ -438,6 +438,87 @@ run_kubectl_server_side_apply_tests() {
   # clean-up
   kubectl delete -f hack/testdata/pod.yaml "${kube_flags[@]:?}"
 
+    # Test apply migration
```

Comment:
[#107980](https://github.com/kubernetes/kubernetes/issues/107980), one of the issues cited as fixed, seems to be about kubectl diff. We're not specifically modifying that command, though I can imagine how the fix to the persisted state might be sufficient. Have you confirmed that specifically? If it is fixed, should we add a regression test for that too?

Reply:
Yes I've also added an integration test to show that after doing the migration to server-side-apply, an object which previously would show a conflict upon a server-side diff now correctly shows the diff.